src/summary/history/allergy-card-level-three-add.component.tsx

```diff
@@ -0,0 +1,338 @@
+import React, { DetailedHTMLProps } from "react";
+import { match } from "react-router";
+import SummaryCard from "../cards/summary-card.component";
+import style from "./allergy-card-level-three.css";
+import {
+  getAllergyAllergenByConceptUuid,
+  getAllergyReaction,
+  savePatientAllergy
+} from "./allergy-intolerance.resource";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { useCurrentPatient } from "@openmrs/esm-api";
+
+const DRUG_ALLERGEN_CONCEPT: string = "162552AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+const ENVIROMENTAL_ALLERGEN_CONCEPT: string =
+  "162554AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+const FOOD_ALLERGEN_CONCEPT: string = "162553AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+
+export function AllergyCardLevelThreeAdd(props: AllergyCardLevelThreeAddProps) {
+  const patientAllergyReactionArray: any[] = [];
+
+  const [codedAllergenUuid, setCodedAllergenUuid] = React.useState(null);
+  const [allergenType, setAllergenType] = React.useState(null);
+  const [allergensArray, setAllergensArray] = React.useState(null);
+  const [allergyReaction, setAllergyReaction] = React.useState();
+  const [disableButton, setDisableButton] = React.useState(true);
+  const [patientAlleryReaction, setPatientAllergyReaction] = React.useState([]);
+  const [comment, setComment] = React.useState("");
+  const [selectedAllergyCategory, setSelectedAllergyCategory] = React.useState(
+    null
+  );
+  const [reactionSeverityUuid, SetReactionSeverityUuid] = React.useState(null);
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+
+  const handleAllergenChange = event => {
+    setAllergensArray(null);
+    setAllergenType(getAllergenType(event.target.value));
+    setSelectedAllergyCategory(event.target.value);
+  };
+
+  function getAllergenType(allergenConcept: string): string {
+    switch (allergenConcept) {
+      case DRUG_ALLERGEN_CONCEPT:
+        return "DRUG";
+      case FOOD_ALLERGEN_CONCEPT:
+        return "FOOD";
+      case ENVIROMENTAL_ALLERGEN_CONCEPT:
+        return "ENVIROMENTAL";
+
+      default:
+        "NO ALLERGEN";
+    }
+  }
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    const patientAllergy: Allergy = {
+      allergenType: allergenType,
+      codedAllergenUuid: codedAllergenUuid,
+      severityUuid: reactionSeverityUuid,
+      comment: comment,
+      reactionsUuid: patientAlleryReaction
+    };
+    const abortController = new AbortController();
+    savePatientAllergy(patientAllergy, patientUuid, abortController)
+      .then(response => {
+        response.status == 201 && navigate();
+      })
+      .catch(createErrorHandler());
+    return () => abortController.abort();
+  };
+
+  function navigate() {
+    window.location.href = `https://openmrs-spa.org/openmrs/spa/patient/${patientUuid}/chart/allergy`;
+  }
+  React.useEffect(() => {
+    if (
+      comment &&
+      allergenType &&
+      codedAllergenUuid != null &&
```

Comment:
I think it's preferred to use the strict version !==

Reply:
Fixed